src/main/java/seedu/address/storage/JsonUniCashStorage.java

```diff
@@ -0,0 +1,79 @@
+package seedu.address.storage;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.exceptions.DataLoadingException;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.commons.util.FileUtil;
+import seedu.address.commons.util.JsonUtil;
+import seedu.address.model.ReadOnlyUniCash;
+
+/**
+ * A class to access UniCash data stored as a json file on the hard disk.
+ */
+public class JsonUniCashStorage implements UniCashStorage {
+
+    private static final Logger logger = LogsCenter.getLogger(JsonUniCashStorage.class);
+
+    private final Path filePath;
+
+    public JsonUniCashStorage(Path filePath) {
+        this.filePath = filePath;
+    }
+
+    public Path getUniCashFilePath() {
+        return filePath;
+    }
+
+    @Override
+    public Optional<ReadOnlyUniCash> readUniCash() throws DataLoadingException {
+        return readUniCash(filePath);
+    }
+
+    /**
+     * Similar to {@link #readUniCash()}.
+     *
+     * @param filePath location of the data. Cannot be null.
+     * @throws DataLoadingException if loading the data from storage failed.
+     */
+    public Optional<ReadOnlyUniCash> readUniCash(Path filePath) throws DataLoadingException {
+        requireNonNull(filePath);
+
+        var jsonUniCash = JsonUtil.readJsonFile(filePath, JsonSerializableUniCash.class);
```

Comment:
Would it be better to replace `var` with `Optional<JsonSerializableUniCash>` for better type safety?

Reply:
Does `var` not also expand to `Option<...>`? I'll check on this, just wanted to cut down on the lengthy variable declarations

Reply:
I see, I did not know about that Java's var could auto infer types. That's pretty neat acutally